### PR TITLE
foot: init at 1.4.4

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

add `foot` a simple and fast terminal emulator for wayland based of `fcft` font rendering.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
@GrahamcOfBorg build foot

File: pkgs/applications/misc/foot/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  nativeBuildInputs = [ meson ninja ncurses scdoc tllist wayland-protocols ];
+  buildInputs = [ fontconfig freetype pixman wayland libxkbcommon fcft pkg-config ];
+
+  # recommended build flags for foot as per INSTALL.md
+  preConfigure = ''
+    export CFLAGS+="-O3 -fno-plt"
+  '';
+
+  mesonFlags = [ "--buildtype=release" "-Db_lto=true" ];
```
Comment (doronbehar) [inline pkgs/applications/misc/foot/default.nix:23]: I *think Nix' meson setup hook sets this build release flags by default. Also, could try enabling tests? https://codeberg.org/dnkl/fcft/src/branch/master/meson.build#L99

File: pkgs/applications/misc/foot/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  src = fetchgit {
+    url = "https://codeberg.org/dnkl/foot.git";
+    rev = "${version}";
+    sha256 = "1cr4sz075v18clh8nlvgyxlbvfkhbsg0qrqgnclip5rwa24ry1lg";
+  };
+
+  nativeBuildInputs = [ meson ninja ncurses scdoc tllist wayland-protocols ];
+  buildInputs = [ fontconfig freetype pixman wayland libxkbcommon fcft pkg-config ];
+
+  # recommended build flags for foot as per INSTALL.md
```
Comment (doronbehar) [inline pkgs/applications/misc/foot/default.nix:18]: A link would be nice.

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/foot/default.nix (hunk 1 above)
Comment (sternenseemann) [reply pkgs/applications/misc/foot/default.nix:24]: I checked and it doesn't seem like it is added (unless its a meson default behavior maybe):
https://github.com/NixOS/nixpkgs/blob/914b9788fd18a0eb908c7bf630cb206f5dabd25e/pkgs/development/tools/build-managers/meson/setup-hook.sh#L9-L17

Enabled the tests for fcft in 5bb0129062e05655d3408ebdc5e3a27141cefd83, good catch!

File: pkgs/applications/misc/foot/default.nix (hunk 2 above)
Comment (sternenseemann) [reply pkgs/applications/misc/foot/default.nix:18]: done

File: pkgs/applications/misc/foot/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchgit, fcft, freetype, pixman, libxkbcommon, fontconfig, wayland
+, meson, ninja, ncurses, scdoc, tllist, wayland-protocols, pkg-config
+}:
+
+stdenv.mkDerivation rec {
+  pname = "foot";
+  version = "1.4.4";
+
+  src = fetchgit {
+    url = "https://codeberg.org/dnkl/foot.git";
```
Comment (doronbehar) [inline pkgs/applications/misc/foot/default.nix:10]: nixpkgs-review keeps failing to me because the server is unreachable...

Comment (doronbehar) [reply pkgs/applications/misc/foot/default.nix:10]: But I _can_ access it via the web.

Comment (sternenseemann) [reply pkgs/applications/misc/foot/default.nix:10]: weird… nixpkgs-review works fine for me. maybe try cloning the repo manually somewhere maybe will yield a helpful error message?!

**Comment by sternenseemann** [comment]
@GrahamcOfBorg build foot fuzzel

**Comment by doronbehar** [comment]
Weird, don't know why it didn't work out for me. Am I supposed to be able to run this on Xorg?

**Comment by sternenseemann** [comment]
No, it is wayland only!

File: pkgs/applications/misc/foot/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchgit, fcft, freetype, pixman, libxkbcommon, fontconfig, wayland
+, meson, ninja, ncurses, scdoc, tllist, wayland-protocols, pkg-config
+}:
```
Comment (doronbehar) [inline pkgs/applications/misc/foot/default.nix:3]:
```suggestion
{ stdenv
, fetchgit
, pkg-config
, fcft
, freetype
, pixman
, libxkbcommon
, fontconfig
, wayland
, meson
, ninja
, ncurses
, scdoc
, tllist
, wayland-protocols
}:
```

File: pkgs/applications/misc/foot/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  pname = "foot";
+  version = "1.4.4";
+
+  src = fetchgit {
+    url = "https://codeberg.org/dnkl/foot.git";
+    rev = "${version}";
+    sha256 = "1cr4sz075v18clh8nlvgyxlbvfkhbsg0qrqgnclip5rwa24ry1lg";
+  };
+
+  nativeBuildInputs = [ meson ninja ncurses scdoc tllist wayland-protocols ];
```
Comment (doronbehar) [inline pkgs/applications/misc/foot/default.nix:15]: Note that `pkg-config` moved here.

```suggestion
  nativeBuildInputs = [
    meson
    ninja
    ncurses
    scdoc
    tllist
    wayland-protocols
    pkg-config
  ];
```

File: pkgs/applications/misc/foot/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  buildInputs = [ fontconfig freetype pixman wayland libxkbcommon fcft pkg-config ];
```
Comment (doronbehar) [inline pkgs/applications/misc/foot/default.nix:16]:
```suggestion
  buildInputs = [
    fontconfig
    freetype
    pixman
    wayland
    libxkbcommon
    fcft
  ];
```

File: pkgs/applications/misc/foot/default.nix [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+  # recommended build flags for foot as per INSTALL.md
+  # https://codeberg.org/dnkl/foot/src/branch/master/INSTALL.md#user-content-release-build
+  preConfigure = ''
+    export CFLAGS+="-O3 -fno-plt"
+  '';
+
+  mesonFlags = [ "--buildtype=release" "-Db_lto=true" ];
```
Comment (doronbehar) [inline pkgs/applications/misc/foot/default.nix:24]:
```suggestion
  mesonFlags = [
    "--buildtype=release"
    "-Db_lto=true"
  ];
```

**Review by doronbehar — CHANGES_REQUESTED** [review]
> No, it is wayland only!

Then I can't help further. The build logs looks good and the tests seem to be running fine so this LGTM. However, I have some personal style nitpicks :)

File: pkgs/applications/misc/foot/default.nix (hunk 5 above)
Comment (sternenseemann) [reply pkgs/applications/misc/foot/default.nix:15]: whoops, pkg-config is in the wrong place indeed

File: pkgs/applications/misc/foot/default.nix [hunk 8]
```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchgit
+, fcft, freetype, pixman, libxkbcommon, fontconfig, wayland
+, meson, ninja, ncurses, scdoc, tllist, wayland-protocols, pkg-config
```
Comment (doronbehar) [inline pkgs/applications/misc/foot/default.nix:3]: You didn't split each of them to it's own line. It's only a personal style preference, but IMO it's much easier to review future diffs this way. Please consider applying my request.

File: pkgs/applications/misc/foot/default.nix [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  version = "1.4.4";
+
+  src = fetchgit {
+    url = "https://codeberg.org/dnkl/foot.git";
+    rev = "${version}";
+    sha256 = "1cr4sz075v18clh8nlvgyxlbvfkhbsg0qrqgnclip5rwa24ry1lg";
+  };
+
+  nativeBuildInputs = [
+    meson ninja ncurses scdoc tllist wayland-protocols pkg-config
```
Comment (doronbehar) [inline pkgs/applications/misc/foot/default.nix:17]: Same here.

File: pkgs/applications/misc/foot/default.nix (hunk 4 above)
Comment (sternenseemann) [reply pkgs/applications/misc/foot/default.nix]: For my taste these just convert horizontal space into vertical space without a real benefit to readability. I reduced horizontal space somewhat and split the inputs by category more obviously.

File: pkgs/applications/misc/foot/default.nix (hunk 7 above)
Comment (doronbehar) [reply pkgs/applications/misc/foot/default.nix:29]: And here.

File: pkgs/applications/misc/foot/default.nix (hunk 8 above)
Comment (sternenseemann) [reply pkgs/applications/misc/foot/default.nix:3]: I don't expect the dependencies to change that much that the indicator of githubs diff rendering what was added and removed in the line wouldn't be good enough for reviewing. Also this is the way arguments and buildInputs are generally formatted in nixpkgs so I'd opt for a similar style generally.

**Pull request merged**